src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs

```diff
@@ -356,7 +365,7 @@ impl HummockVersionUpdateExt for HummockVersion {
                     });
                 add_ssts_to_sub_level(
                     target_l0,
-                    target_level_idx,
+                    insert_hint,
```

Comment:
> Instead of passing the `insert_hint` to `add_ssts_to_sub_level`, we can do `match insert_hint` here and call `add_ssts_to_sub_level` when matched `Ok(idx)` and `insert_new_sub_level` when matched `Err(idx)`

ok.